gui/exported.go

```diff
@@ -88,3 +90,61 @@ func (app *App) GoLeaveNetwork(networkName string) (any, error) {
 	}
 	return nil, fmt.Errorf("%w: "+errMsgsBuilder.String(), err)
 }
+
+// App.GoGetRecentServerNames returns names of all known (joined) servers
+func (app *App) GoGetRecentServerNames() ([]string, error) {
+	serverNames := []string{}
+	for name := range config.Servers {
+		name := name
+		serverNames = append(serverNames, name)
+	}
+	return serverNames, nil
+}
+
+// App.GoJoinNetworkBySso joins a network by SSO
+func (app *App) GoJoinNetworkBySso(serverName, networkName string) (any, error) {
+	flags := viper.New()
+	flags.Set("server", serverName)
+	flags.Set("network", networkName)
+
+	cmd.InitConfig()
+	err := functions.Join(flags)
+	if err != nil {
+		fmt.Println(err)
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+// App.GoJoinNetworkBySso joins a network by SSO
```

Comment:
comment needs updating copy/paste errror?

Reply:
perhaps we need to consolidate after this merge to ensure it works..

Reply:
agreed

Reply:
resolved